### PR TITLE
WIP:  Multi-architecture build for for the nodejs actions

### PR DESCRIPTION
**Pull request by jonpspri**
These changes are intended to guide the OpenWhisk actions toward being available as multi-architecture meanifests within the Docker Hub repository.  Their availability in that form means that OpenWhisk users running on non-x86 platforms (e.g. ARM, s390x, Power, etc.).

To achieve multi-platform build, I've had (or chosen) to make several changes to the build process.  The most significant is the adoption of the 'gradle-docker-plugin' over the current 'gradle/docker.gradle' code.  The primary reasons for adopting the plugin were to reduce future maintenance in keeping up with Docker, pre-existing TLS integration for remote repositories and to use a Plugin where docker addressing information (and encryption information) were held in a defined Project Extension rather than passed as arguments.

Docker and Docker Registry configuration now happens in a docker-local.gradle file (sample included in build); image configuration happens in 'settings.gradle' to ensure that all necessary information is available at the beginning of build configuration.  The root 'build.gradle' file has been completely re-written, and (in the spirit of old-time Makefiles) as much universal logic as could be about product structure has been made visible within that file.  I also did my best to comment heavily throughout that file.

Backwards compatibility with the 'distDocker' tasks has been maintained as best as manageable; however registry login information must now be kept in the 'docker-local.gradle' file because the underlying library no longer uses the docker command line to push images.

Once this is refined, my intention would be to use the same framework (and ideally root build.gradle file) in all other actions; however this action is the one most immediately needed to support work I'm doing to make OpenWhisk available on the S390X platform.

I'll be hanging out on Slack and the usual other channels to discuss and review.  I expect we'll go through a few iterations to make this workable.

Cheers!
Jonathan

### CODE REVIEW

File: core/nodejsActionBase/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM buildpack-deps:trusty-curl
+FROM buildpack-deps:xenial-curl
```
Comment (csantanapr) [inline core/nodejsActionBase/Dockerfile]: It’s required to change to Xenia? Can we keep trusty?

Comment (jonpspri) [reply core/nodejsActionBase/Dockerfile]: Xenial is necessary for the s390x port -- trusty does not have a supported port to s390x, I chose xenial as the next LTR.  I can add logic to make it conditional depending on platform -- Groovy templates as part of Gradle are a powerful thing, or we can start the inevitable move to Xenial.  Let me know which you prefer.

**Comment by jonpspri** [comment]
I added support for conditional selection of underlying ubuntu image (trusty vs. xenial).  Unfortunately, until gradle-docker-plugin 3.2.3 is released, it needs a build from the gradle-docker-plugin github to work.  My thinking is to sit on this PR for a bit until the next release -- the PR for the needed changes has been accepted upstream by the gradle-docker-plugin maintainers.

**Comment by jonpspri** [comment]
I'm marking this WIP pending the next release of the gradle-docker-plugin with DockerInfo JSON output capability that our build scripts will need.

**Comment by jonpspri** [comment]
Closing because the commit chain has gotten too unweildy.  Will reopen another PR when ready for commit.  Work continues.

**Pull request closed without merging**